File: InternetScanner/languages/csharp/InternetScanner.cs
using System.Net;
namespace BizX.InternetScanner;
public static class InternetScanner {
 public static string Classify(string ip){ var a=IPAddress.Parse(ip); var b=a.GetAddressBytes(); if(a.AddressFamily==System.Net.Sockets.AddressFamily.InterNetwork && (b[0]==10 || (b[0]==172&&b[1]>=16&&b[1]<=31)||(b[0]==192&&b[1]==168))) return "local/intranet"; if(IPAddress.IsLoopback(a)||a.IsIPv6LinkLocal) return "local/intranet"; return "public"; }
 public static bool Authorized(string ip,ISet<string> allow)=>Classify(ip)!="public"||allow.Contains(ip);
}

File: InternetScanner/languages/swift/InternetScanner.swift
import Foundation
public enum InternetScanner {
 public static func classify(_ ip:String)->String { guard let a=IPv4Address(ip) else{return "public"}; let o=a.rawValue; if o[0]==10 || (o[0]==172 && (16...31).contains(o[1])) || (o[0]==192 && o[1]==168) || o[0]==127 { return "local/intranet" }; return "public" }
 public static func authorized(_ ip:String, allow:Set<String>)->Bool { classify(ip) != "public" || allow.contains(ip) }
}
